File: src/ds/python/stack.py
class Stack:

    stack = []

    def push(self, item):
        self.stack.append(item)

    def pop(self):
        return self.stack.pop()

    def isEmpty(self):
        return  len(self.stack) == 0

    def peek(self):
        return self.stack[len(self.stack) - 1]





stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)
stack.push(5)

# print('as')
print(stack.stack)
stack.pop()
print(stack.stack)
print(stack.peek())

File: src/DataStructures/stack.php
#!/usr/bin/php
<?php

class Stack {
    protected $stack = [];
    
    // add to top of stack
    public function push($item) {
        $this->stack[] = $item;
    }
    
    // remove from top of stack
    public function pop() {
        return array_pop($this->stack);
    }
    
    // return top of stack
    public function peek() {
        return $this->stack[count($this->stack) - 1];
    }
    
    // return true if stack is empty
    public function isEmpty() {
        return count($this->stack) == 0;
    }

    // note : this is only for debugging
    public function get() {
        return $this->stack;
    }
}

$stack = new Stack();
$stack->push(1);
$stack->push(3);
$stack->push(2);
$stack->push(4);
$stack->push(5);

echo "Top = " . $stack->peek() . PHP_EOL;
$stack->pop();
echo "Top = " . $stack->peek() . PHP_EOL;